Framework/PythonInterface/plugins/algorithms/HB2AReduce.py

```diff
@@ -193,6 +274,26 @@ def PyExec(self):
 
         self.add_metadata(outWS, metadata, data)
 
+        # save reduced workspace to requested format
+        save_data = self.getProperty("SaveData").value
+        if save_data:
+            outputdir = self.getProperty("OutputDirectory").value
+            outputdir = outputdir if outputdir != "" else f"/HFIR/HB2A/IPTS-{metadata['proposal']}/shared"
+            _outputfunc = {
+                'XYE': SaveFocusedXYE,
+                'GSAS': SaveGSS
+            }[self.getProperty('OutputFormat').value]
+            _outputext = {
+                "XYE": 'dat',
+                "GSAS": 'gss',
+            }[self.getProperty('OutputFormat').value]
+            outputbase = os.path.join(outputdir, self.getProperty("OutputWorkspace").value)
```

Comment:
```suggestion
            outputbase = os.path.join(outputdir, self.getPropertyValue("OutputWorkspace"))
```

This will then return the "OutputWorkspace" as a string instead of a reference to the workspace

Reply:
actually the property got recast into a Workspace2D a few lines above.  I cached the string value at the beginning to get the correct output filename.